model/src/main/java/org/eclipse/rdf4j/model/util/Models.java

```diff
@@ -663,55 +696,15 @@ private static boolean statementsMatch(Statement st1, Statement st2, Map<Resourc
 		Resource subj1 = st1.getSubject();
 		Resource subj2 = st2.getSubject();
 
-		if (isBlank(subj1) && isBlank(subj2)) {
-			Resource mappedBNode = bNodeMapping.get(subj1);
-
-			if (mappedBNode != null) {
-				// bNode 'subj1' was already mapped to some other bNode
-				if (!subj2.equals(mappedBNode)) {
-					// 'subj1' and 'subj2' do not match
-					return false;
-				}
-			} else {
-				// 'subj1' was not yet mapped. we need to check if 'subj2' is a
-				// possible mapping candidate
-				if (bNodeMapping.containsValue(subj2)) {
-					// 'subj2' is already mapped to some other value.
-					return false;
-				}
-			}
-		} else {
-			// subjects are not (both) bNodes
-			if (!subj1.equals(subj2)) {
-				return false;
-			}
+		if (bnodeValueMatching(bNodeMapping, subj1, subj2)) {
```

Comment:
Extracted out this method, mostly because intellij told me that there were 3 duplicates. Also gave a small performance improvement, since the smaller method is easier to optimise for the jvm.